File: windows/Liferay.SDK/Service/V73/Sharingentry/SharingentryService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Sharingentry
{
	public class SharingentryService : ServiceBase
	{
		public SharingentryService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> AddOrUpdateSharingEntryAsync(long toUserId, long classNameId, long classPK, long groupId, bool shareable, IEnumerable<object> sharingEntryActions, long expirationDate, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("toUserId", toUserId);
			_parameters.Add("classNameId", classNameId);
			_parameters.Add("classPK", classPK);
			_parameters.Add("groupId", groupId);
			_parameters.Add("shareable", shareable);
			_parameters.Add("sharingEntryActions", sharingEntryActions);
			_parameters.Add("expirationDate", expirationDate);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/sharing.sharingentry/add-or-update-sharing-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> DeleteSharingEntryAsync(long sharingEntryId, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("sharingEntryId", sharingEntryId);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/sharing.sharingentry/delete-sharing-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateSharingEntryAsync(long sharingEntryId, IEnumerable<object> sharingEntryActions, bool shareable, long expirationDate, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("sharingEntryId", sharingEntryId);
			_parameters.Add("sharingEntryActions", sharingEntryActions);
			_parameters.Add("shareable", shareable);
			_parameters.Add("expirationDate", expirationDate);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/sharing.sharingentry/update-sharing-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddSharingEntryAsync(long toUserId, long classNameId, long classPK, long groupId, bool shareable, IEnumerable<object> sharingEntryActions, long expirationDate, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("toUserId", toUserId);
			_parameters.Add("classNameId", classNameId);
			_parameters.Add("classPK", classPK);
			_parameters.Add("groupId", groupId);
			_parameters.Add("shareable", shareable);
			_parameters.Add("sharingEntryActions", sharingEntryActions);
			_parameters.Add("expirationDate", expirationDate);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/sharing.sharingentry/add-sharing-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/sharingentry/SharingentryService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.sharingentry;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class SharingentryService extends BaseService {

	public SharingentryService(Session session) {
		super(session);
	}

	public JSONObject addOrUpdateSharingEntry(long toUserId, long classNameId, long classPK, long groupId, boolean shareable, JSONArray sharingEntryActions, long expirationDate, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("classNameId", classNameId);
			_params.put("classPK", classPK);
			_params.put("expirationDate", expirationDate);
			_params.put("groupId", groupId);
			_params.put("shareable", shareable);
			_params.put("sharingEntryActions", checkNull(sharingEntryActions));
			_params.put("toUserId", toUserId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/sharing.sharingentry/add-or-update-sharing-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject addSharingEntry(long toUserId, long classNameId, long classPK, long groupId, boolean shareable, JSONArray sharingEntryActions, long expirationDate, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("classNameId", classNameId);
			_params.put("classPK", classPK);
			_params.put("expirationDate", expirationDate);
			_params.put("groupId", groupId);
			_params.put("shareable", shareable);
			_params.put("sharingEntryActions", checkNull(sharingEntryActions));
			_params.put("toUserId", toUserId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/sharing.sharingentry/add-sharing-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject deleteSharingEntry(long sharingEntryId, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("sharingEntryId", sharingEntryId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/sharing.sharingentry/delete-sharing-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateSharingEntry(long sharingEntryId, JSONArray sharingEntryActions, boolean shareable, long expirationDate, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("expirationDate", expirationDate);
			_params.put("shareable", shareable);
			_params.put("sharingEntryActions", checkNull(sharingEntryActions));
			_params.put("sharingEntryId", sharingEntryId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/sharing.sharingentry/update-sharing-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}